helm_drydock/drivers/node/maasdriver/driver.py

```diff
@@ -0,0 +1,306 @@
+# Copyright 2017 AT&T Intellectual Property.  All other rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import helm_drydock.error as errors
+import helm_drydock.config as config
+import helm_drydock.drivers as drivers
+import helm_drydock.objects.fields as hd_fields
+import helm_drydock.objects.task as task_model
+
+from helm_drydock.drivers.node import NodeDriver
+from .api_client import MaasRequestFactory
+import helm_drydock.drivers.node.maasdriver.models.fabric as maas_fabric
+import helm_drydock.drivers.node.maasdriver.models.vlan as maas_vlan
+import helm_drydock.drivers.node.maasdriver.models.subnet as maas_subnet
+
+class MaasNodeDriver(NodeDriver):
+
+    def __init__(self, **kwargs):
+        super(MaasNodeDriver, self).__init__(**kwargs)
+	
+        self.driver_name = "maasdriver"
+        self.driver_key = "maasdriver"
+        self.driver_desc = "MaaS Node Provisioning Driver"
+
+        self.config = config.DrydockConfig.node_driver[self.driver_key]
+
+    def execute_task(self, task_id):
+        task = self.state_manager.get_task(task_id)
+
+        if task is None:
+            raise errors.DriverError("Invalid task %s" % (task_id))
+
+        if task.action not in self.supported_actions:
+            raise errors.DriverError("Driver %s doesn't support task action %s"
+                % (self.driver_desc, task.action))
+
+        if task.action == hd_fields.OrchestratorAction.ValidateNodeServices:
```

Comment:
instead of overriding `execute_task` I would of expected this to stay generic, avoid needing to be overridden, and depending on the task action, call to (mock code):

```
f = getattr(self, task.action)
f(*args, **kwargs)
```

so that `MaasNodeDriver` simply implements its version of

```
def validate_node_services(...)
```

This way, common logic that is not driver dependent can remain within execute_task, as it exists in the base driver.